File: 231213/괄호 쌍 만들어주기 3/pair-parentheses-3.js
const fs = require('fs');
const input = () => fs.readFileSync('/dev/stdin').toString()

const parString = input();
let answer = 0;

for(let i = 0; i < parString.length; i++){
    const par = parString[i];

    if(par === '('){
        for(let j = i + 1; j < parString.length; j++){
            const par2 = parString[j];

            if(par2 === ')'){
                answer += 1;
            }
        }
    }
}

console.log(answer);

File: 231213/괄호 쌍 만들어주기 3/pair-parentheses-3.py
# 첫 번째 여는 괄호를 찾는다
# 그 뒤에서 여는 괄호를 찾는다.
# 찾을 때마다 개수를 추가한다.

parList = input()

count = 0

for i in range(len(parList)):
    par = parList[i]

    if par == '(':
        for j in range(i+1, len(parList)):
            par2 = parList[j]

            if par2 == ")":
                count += 1

print(count)